### ISSUE TITLE
distributed cannot be first imported by a non-main thread

### ISSUE THREAD

**Issue by hmaarrfk** [dask/distributed#5587]
```python
from threading import Thread
import traceback
def import_dask_distributed():
    try:
        import dask.distributed
    except Exception as e:
        traceback.print_exc()
x = Thread(target=import_dask_distributed)
x.start()
```

**What happened**:
Running the above code results in the following output:
```
Traceback (most recent call last):
  File "/tmp/ipykernel_76351/1031499154.py", line 5, in import_dask_distributed
    import dask.distributed
  File "/home/mark/mambaforge/envs/dev/lib/python3.9/site-packages/dask/distributed.py", line 11, in <module>
    from distributed import *
  File "/home/mark/mambaforge/envs/dev/lib/python3.9/site-packages/distributed/__init__.py", line 7, in <module>
    from .actor import Actor, ActorFuture
  File "/home/mark/mambaforge/envs/dev/lib/python3.9/site-packages/distributed/actor.py", line 5, in <module>
    from .client import Future
  File "/home/mark/mambaforge/envs/dev/lib/python3.9/site-packages/distributed/client.py", line 59, in <module>
    from .batched import BatchedSend
  File "/home/mark/mambaforge/envs/dev/lib/python3.9/site-packages/distributed/batched.py", line 10, in <module>
    from .core import CommClosedError
  File "/home/mark/mambaforge/envs/dev/lib/python3.9/site-packages/distributed/core.py", line 28, in <module>
    from .comm import (
  File "/home/mark/mambaforge/envs/dev/lib/python3.9/site-packages/distributed/comm/__init__.py", line 25, in <module>
    _register_transports()
  File "/home/mark/mambaforge/envs/dev/lib/python3.9/site-packages/distributed/comm/__init__.py", line 17, in _register_transports
    from . import inproc, tcp, ws
  File "/home/mark/mambaforge/envs/dev/lib/python3.9/site-packages/distributed/comm/tcp.py", line 387, in <module>
    class BaseTCPConnector(Connector, RequireEncryptionMixin):
  File "/home/mark/mambaforge/envs/dev/lib/python3.9/site-packages/distributed/comm/tcp.py", line 389, in BaseTCPConnector
    _resolver = netutil.ExecutorResolver(close_executor=False, executor=_executor)
  File "/home/mark/mambaforge/envs/dev/lib/python3.9/site-packages/tornado/util.py", line 288, in __new__
    instance.initialize(*args, **init_kwargs)
  File "/home/mark/mambaforge/envs/dev/lib/python3.9/site-packages/tornado/netutil.py", line 427, in initialize
    self.io_loop = IOLoop.current()
  File "/home/mark/mambaforge/envs/dev/lib/python3.9/site-packages/tornado/ioloop.py", line 263, in current
    loop = asyncio.get_event_loop()
  File "/home/mark/mambaforge/envs/dev/lib/python3.9/asyncio/events.py", line 642, in get_event_loop
    raise RuntimeError('There is no current event loop in thread %r.'
RuntimeError: There is no current event loop in thread 'Thread-10'.
```

**What you expected to happen**:

It not to crash 😄 

**Anything else we need to know?**:

I think it is related to the recent change in:
https://github.com/dask/distributed/pull/5531

**Environment**:

- Dask version: `'2021.12.0'`
- Python version: `'3.9.7 | packaged by conda-forge | (default, Sep 29 2021, 19:20:46) \n[GCC 9.4.0]'`
- Operating System: Ubuntu 21.10
- Install method (conda, pip, source): conda (conda-forge)

**Comment by hmaarrfk** [comment]
xref: https://github.com/dask/dask/issues/8466

**Comment by jrbourbeau** [comment]
Thanks for reporting @hmaarrfk. cc @jcrist for visibility

**Comment by jcrist** [comment]
Ah, excellent, I think this is likely the cause of our periodic test failures. Should be a quick fix, thanks for raising!

**Comment by jcrist** [comment]
Should be fixed by #5593. This was actually a long-standing bug (kinda - it was wrong, but didn't matter), but since we were magically creating event loops before it was hidden until now. Thanks for tracking this down!
